neqo-http3/src/connection.rs

```diff
@@ -150,26 +150,33 @@ impl Http3Connection {
         !self.streams_have_data_to_send.is_empty()
     }
 
-    /// Call `send` for all streams that need to send data.
-    pub fn process_sending(&mut self, conn: &mut Connection) -> Res<()> {
-        // check if control stream has data to send.
-        self.control_stream_local
-            .send(conn, &mut self.recv_streams)?;
-
+    fn send_non_control_streams(&mut self, conn: &mut Connection) -> Res<()> {
         let to_send = mem::take(&mut self.streams_have_data_to_send);
         for stream_id in to_send {
-            let mut remove = false;
-            if let Some(s) = &mut self.send_streams.get_mut(&stream_id) {
+            let done = if let Some(s) = &mut self.send_streams.get_mut(&stream_id) {
                 s.send(conn)?;
                 if s.has_data_to_send() {
                     self.streams_have_data_to_send.insert(stream_id);
```

Comment:
Just noticed that this should be called streams_with_data_to_send, streams_with_pending_data, or similar.